docs/config.md

```diff
@@ -1,76 +1,182 @@
+| `attestation_token_type` | String       | Attestation token broker type. Available only when the `resource` feature is enabled. Valid values: `CoCo` | No       | `CoCo`               |
+| `sockets`                | String array | One or more sockets to listen on.                                                                          | No       | `["127.0.0.1:8080"]` |
+| `insecure_api`           | Boolean      | Enable KBS insecure APIs such as Resource Registration without JWK verification.                           | No       | `false`              |
+| `insecure_http`          | Boolean      | Don't use TLS for the KBS HTTP endpoint.                                                                   | No       | `false`              |
+| `timeout`                | Integer      | HTTP session timeout in minutes.                                                                           | No       | `5`                  |
+| `private_key`            | String       | Path to a private key file to be used for HTTPS.                                                           | No       | -                    |
+| `certificate`            | String       | Path to a certificate file to be used for HTTPS.                                                           | No       | -                    |
+| `auth_public_key`        | String       | Path to a public key file to be used for authenticating the resource registration endpoint token (JWT).    | No       | -                    |
+
+### Repository Configuration
+
+The following properties can be set under the `repository_config` section.
+
+This section is **optional**. When omitted, a default configuration is used.
+
+Repository configuration is **specific to a repository type**. See the following sections for
+type-specific properties.
+
+>This section is available only when the `resource` feature is enabled.
+
+| Property | Type   | Description                                           | Required | Default   |
+|----------|--------|-------------------------------------------------------|----------|-----------|
+| `type`   | String | The resource repository type. Valid values: `LocalFs` | Yes      | -         |
+
+**`LocalFs` Properties**
+
+| Property   | Type   | Description                     | Required | Default                                             |
+|------------|--------|---------------------------------|----------|-----------------------------------------------------|
+| `dir_path` | String | Path to a repository directory. | No       | `/opt/confidential-containers/kbs/repository`       |
+
+### Native Attestation
+
+The following properties can be set under the `as_config` section.
+
+This section is **optional**. When omitted, a default configuration is used.
+
+>This section is available only when one or more of the following features are enabled:
+>`coco-as-builtin`, `coco-as-builtin-no-verifier`
```

Comment:
I actually took the AS config as a given, which in retrospect might not make a lot of sense given that both components (KBS and AS) are maintained by the same people. I just wanted to focus on KBS for now.

AS seems to be using a JSON file for configuration at the moment. Maybe it could make sense to do a similar refactoring to AS, too, and then we could maybe import the AS config type in KBS rather than defining it locally.

I've attempted to consolidate *all* KBS config into a single schema, and therefore it seemed strange to me to leave just the AS-related parameters in a JSON file. On the other hand, if that JSON file actually doesn't belong to KBS but to AS, maybe it does make sense to leave it as it is.

What do you think?

Reply:
Oh, actually maybe we *can* consume the AS config struct from the AS repo since that type exists regardless of whether AS unmarshals from JSON or from another format. Let me see if we can reuse.

Reply:
Actually we already *are* consuming the config struct from that repo and don't redefine it locally. Sorry for the confusion.

Let me try to reorganize my thoughts:

>How does configuration work when we run the AS separately?

You are asking how we should configure AS when KBS is talking to AS over gRPC. Correct?

If so, I think AS can use whatever configuration method since KBS is agnostic about how AS is configured, but from a UX perspective it might make sense to have both services use a similar approach.

Reply:
Yeah, in theory the AS could even be used as a standalone component in other projects or deployments. I guess the config for it is kind of orthogonal to this PR, but maybe we can look at something similar for the AS in the future.